OneSignalSDK/onesignal/src/main/java/com/onesignal/NotificationOpenedProcessor.java

```diff
@@ -74,7 +74,13 @@ private static void handleDismissFromActionButtonPress(Context context, Intent i
       // Pressed an action button, need to clear the notification and close the notification area manually.
       if (intent.getBooleanExtra("action_button", false)) {
          NotificationManagerCompat.from(context).cancel(intent.getIntExtra(BUNDLE_KEY_ANDROID_NOTIFICATION_ID, 0));
-         context.sendBroadcast(new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS));
+
+         //Close notification drawer user override targetSdkVersion to  be lower than Android 11
+         // See https://developer.android.com/about/versions/12/behavior-changes-all#close-system-dialogs
+         int targetSdkVersion = context.getApplicationContext().getApplicationInfo().targetSdkVersion;
+         if (targetSdkVersion < Build.VERSION_CODES.S ) {
```

Comment:
This is checking for the `targetSdkVersion` which only accounts for what was set in the app, not the device the app is running on. To get this you can use `Build.VERSION.SDK_INT`.

Reply:
I was under the impression  it depends on OS the device is running on at first. But this behavior change depends only on the `targetSdkVersion`. If you target 12 or higher, regardless of whatever OS you're running, it will throw an `SecurityException`. If you target less than 12, Android 12 will do nothing and log an error whereas it will function as normal on previous devices. Since it's not a crash, we can safely run this code without making an additional check for OS version.

Reply:
I think it'd be better to an OS check rather than `targetSdkVersion`. I was under the impression that it only depends on `targetSdkVersion`.